solutions/devsprint-michel-ferreira-1/src/main/kotlin/com/devpass/challengehexagonal/resources/repository/database/client/TransactionSpringDataRepository.kt

```diff
@@ -3,10 +3,19 @@ package com.devpass.challengehexagonal.resources.repository.database.client
 import com.devpass.challengehexagonal.resources.repository.entity.TransactionEntity
 import java.time.LocalDateTime
 import org.springframework.data.repository.CrudRepository
+import java.math.BigDecimal
+import java.time.LocalDateTime
 
 interface TransactionSpringDataRepository : CrudRepository<TransactionEntity, Long> {
 
     fun findByAccountId(accountId: Long): List<TransactionEntity>
+
```

Comment:
@Pedroajs Essa é a função que o Sprint Data irá executar, para que ele entenda qual a query que ele precisa executar no Banco precisamos seguir o padrão do Spring Data:

fun existsByAccountIdAndValueAndEstablishmentAndTransactionDateBetween(
        accountId: Long,
        value: BigDecimal
        establishment: String,
        startDate: LocalDateTime,
        endDate: LocalDateTime
    ) : Boolean
    
Dessa forma ele vai conseguir montar a query ;)

existsByAccountIdAndValueAndEstablishmentAndTransactionDateBetween
exists -> internamente fará um count e retornará true ou false
by -> representa nossa cláusula 'Where'
accountId -> filtra pela accountId (account_id = ?)
AndValue -> filtra pelo value (value = ?)
AndEstablishment  -> filtra por estabelecimento (establishment = ?)
AndTransactionDateBetween -> cria o filtro da data (and transaction_date between ? and ?)

Reply:
Obrigado, Michel! Agora entendi direitinho